paperqa/config.py

```diff
@@ -0,0 +1,413 @@
+        return v
+
+    @field_validator("select")
+    @classmethod
+    def check_select(cls, v: str) -> str:
+        if not set(get_formatted_variables(v)).issubset(
+            set(get_formatted_variables(select_paper_prompt))
+        ):
+            raise ValueError(
+                f"Select prompt can only have variables: {get_formatted_variables(select_paper_prompt)}"
+            )
+        return v
+
+    @field_validator("pre")
+    @classmethod
+    def check_pre(cls, v: str | None) -> str | None:
+        if v is not None and set(get_formatted_variables(v)) != {"question"}:
+            raise ValueError("Pre prompt must have input variables: question")
+        return v
+
+    @field_validator("post")
+    @classmethod
+    def check_post(cls, v: str | None) -> str | None:
+        if v is not None:
+            # kind of a hack to get list of attributes in answer
+            attrs = set(Answer.model_fields.keys())
+            if not set(get_formatted_variables(v)).issubset(attrs):
+                raise ValueError(f"Post prompt must have input variables: {attrs}")
+        return v
+
+
+class AgentSettings(BaseModel):
+    agent_llm: str = Field(
+        default="gpt-4o-2024-08-06",
+        description="Model to use for agent",
+    )
+
+    agent_system_prompt: str | None = Field(
+        # Matching https://github.com/langchain-ai/langchain/blob/langchain%3D%3D0.2.3/libs/langchain/langchain/agents/openai_functions_agent/base.py#L213-L215
+        default="You are a helpful AI assistant.",
+        description="Optional system prompt message to precede the below agent_prompt.",
+    )
+
+    # TODO: make this prompt more minimalist, instead improving tool descriptions so
+    # how to use them together can be intuited, and exposing them for configuration
+    agent_prompt: str = (
+        "Answer question: {question}"
+        "\n\nSearch for papers, gather evidence, collect papers cited in evidence then re-gather evidence, and answer."
+        " Gathering evidence will do nothing if you have not done a new search or collected new papers."
+        " If you do not have enough evidence to generate a good answer, you can:"
+        "\n- Search for more papers (preferred)"
+        "\n- Collect papers cited by previous evidence (preferred)"
+        "\n- Gather more evidence using a different phrase"
+        "\nIf you search for more papers or collect new papers cited by previous evidence,"
+        " remember to gather evidence again."
+        " Once you have five or more pieces of evidence from multiple sources, or you have tried a few times, "
+        "call {gen_answer_tool_name} tool. The {gen_answer_tool_name} tool output is visible to the user, "
+        "so you do not need to restate the answer and can simply terminate if the answer looks sufficient. "
+        "The current status of evidence/papers/cost is {status}"
+    )
+    paper_directory: str | os.PathLike = Field(
+        default=Path.cwd(),
```

Comment:
Can we add to this field's description explaining why we have a relative path? My understanding is it enables the CLI to work regardless of the starting directory

Cc @mskarlin